### ISSUE TITLE
Inverse matches should not reorder the remaining results

### ISSUE THREAD

**Issue by dimo414** [junegunn/fzf#1458]
- Category
    - [x] fzf binary
    - [ ] fzf-tmux script
    - [ ] Key bindings
    - [ ] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.
- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Windows Subsystem for Linux
    - [ ] Etc.
- Shell
    - [x] bash
    - [ ] zsh
    - [ ] fish

Using inverse matches (`!`) causes the resulting list to be reordered, seemingly by string length. This isn't very helpful, especially in cases where the original order was relevant (in my case the input was ordered by last-access date, and I wanted to filter out items with a certain substring, but lost the ordering in the process).

For example:

```shell
$ printf '%s\n' aaaaa b ccc BAD | fzf
---
  BAD
  ccc
  b
> aaaaa
  4/4
> _
---
  aaaaa
> ccc
  b
  3/4
> !bad
```

**Comment by junegunn** [comment]
Thanks for the suggestion. Fixed in 5624a89.

**Comment by dimo414** [comment]
Awesome! Thanks for the quick turnaround!
